crates/proto/src/lib.rs

```diff
@@ -81,23 +81,26 @@ pub mod iocompat {
     use std::task::{Context, Poll};
 
     use futures_io::{AsyncRead, AsyncWrite};
-    use tokio::io::{AsyncRead as AsyncRead02, AsyncWrite as AsyncWrite02};
+    use tokio::io::{AsyncRead as TokioAsyncRead, AsyncWrite as TokioAsyncWrite, ReadBuf};
 
     /// Conversion from `tokio::io::{AsyncRead, AsyncWrite}` to `std::io::{AsyncRead, AsyncWrite}`
-    pub struct AsyncIo02As03<T>(pub T);
+    pub struct AsyncIoTokioAsStd<T: TokioAsyncRead + TokioAsyncWrite>(pub T);
 
-    impl<T> Unpin for AsyncIo02As03<T> {}
-    impl<R: AsyncRead02 + Unpin> AsyncRead for AsyncIo02As03<R> {
+    impl<T: TokioAsyncRead + TokioAsyncWrite> Unpin for AsyncIoTokioAsStd<T> {}
```

Comment:
I feel like we should probably have a `T: Unpin` bound here?

Reply:
agreed, though it doesn't seem to require it. I'll add that back.

Reply:
This should be done in the resent changes.